### PR TITLE
[ALLUXIO-2119] Sleeping timer test cleanup

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2119

This change applies dependency injection so that we don't need to use sleeping or whitebox to test the sleeping timer.

### CODE REVIEW

File: core/common/src/main/java/alluxio/time/Sleeper.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+package alluxio.time;
+
+/**
+ * An interface for a utility which provides a sleep method.
+ */
+public interface Sleeper {
+
+  /**
+   * @param millis the number of milliseconds to sleep for
```
Comment (gpang) [inline core/common/src/main/java/alluxio/time/Sleeper.java]: NIT: should this have a comment as well?

File: core/common/src/test/java/alluxio/clock/FakeClock.java [hunk 2]
```diff
@@ -14,13 +14,13 @@
 /**
  * A modifiable clock useful for testing.
  */
-public final class TestClock implements Clock {
-  private long mTime;
+public final class FakeClock implements Clock {
```
Comment (gpang) [inline core/common/src/test/java/alluxio/clock/FakeClock.java]: Isn't this more of a modifiable clock or manual clock? "fake" sounds a bit strange.

**Comment by gpang** [comment]
Thanks @aaudiber . I left a few minor comments, looks good after that.

File: core/common/src/main/java/alluxio/time/Sleeper.java (hunk 1 above)
Comment (aaudiber) [inline core/common/src/main/java/alluxio/time/Sleeper.java]: Done

File: core/common/src/test/java/alluxio/clock/FakeClock.java (hunk 2 above)
Comment (aaudiber) [inline core/common/src/test/java/alluxio/clock/FakeClock.java]: Changed to `ManualClock`

**Comment by aaudiber** [comment]
@gpang Comments addressed, PTAL

**Comment by gpang** [comment]
LGTM

**Pull request merged**